### DISCUSSION TITLE
A "Better than nothing" VSCode Extension for Nue Language Support

### DISCUSSION THREAD

**Discussion by yyz945947732** [nuejs/nue#65] [Show and tell]
I hope one day someone can make a complete and intelligent extension like [vetur](https://github.com/vuejs/vetur) for Nue JS. At present, there are no relevant extension. So I just made a humble simple one: 

[Nue Language Support](https://marketplace.visualstudio.com/items?itemName=yaoyuanzhang.nue&ssr=false)

If you install this extension it will:

- ~~set your file associations for "*.nue" to html by default.~~ Activate syntax support for `.nue` files.
- highlight `@xxx` 、`:xxx` 、`{ xxx }` 、`{{ xxx }}` ... stuff.
  
  ![image](https://github.com/nuejs/nuejs/assets/47586954/43fe0fbf-469e-4056-b78c-a1c423683c83)

- The following syntax will no longer report errors.
   
  ![image](https://github.com/nuejs/nuejs/assets/47586954/a4a377f0-c704-4528-b2c5-411441339ddc)
- get [snippets](https://github.com/yyz945947732/vscode-nue#snippets) to write code more easily.
    
  ![image](https://github.com/nuejs/nuejs/assets/47586954/2074766c-3576-4841-b703-231b52eb59d6)

  ![image](https://github.com/nuejs/nuejs/assets/47586954/2c1b6442-db50-42fb-bc0e-3183e23bced5)

- get [auto-completion](https://github.com/yyz945947732/vscode-nue#auto-completion) and syntax information when you want to code `:if`、`:for`... stuff.

  ![image](https://github.com/nuejs/nuejs/assets/47586954/6495eed8-ac31-4311-8429-e6182744a810)

  ![image](https://github.com/nuejs/nuejs/assets/47586954/82a5e854-fcfa-42c6-9685-d289f5930294)

- ~~The above will affect all html type files, not just `.nue` files, you can still use the `.html` file suffix.~~

Basically the above. By now, it probably has a lot of bugs, but it should be better than code without any extension. If you use it and find problems please let me know. 

@tipiirai

**Comment by tipiirai** [comment]
Awesome! This is pretty complete already. I'll put a link to this on the website somewhere.

**Reply by tipiirai** [reply]
Now linked on the Getting started page: https://nuejs.org/docs/nuejs/getting-started.html

ps: I pushed a new blog entry as well: https://nuejs.org/blog/rethinking-reactivity/

**Comment by seneca** [comment]
This is awesome. Thank you!!!

**Comment by felix-bohlin** [comment]
Can recommend checking Volar.js out!
https://volarjs.dev/
